src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc

```diff
@@ -362,7 +375,53 @@ void PerformAudioPassThruRequest::FinishTTSSpeak() {
 
 bool PerformAudioPassThruRequest::IsWaitingHMIResponse() {
   LOG4CXX_AUTO_TRACE(logger_);
-  return awaiting_tts_speak_response_ || awaiting_ui_response_;
+  return GetInterfaceAwaitState(HmiInterfaces::HMI_INTERFACE_TTS) ||
+         GetInterfaceAwaitState(HmiInterfaces::HMI_INTERFACE_UI);
+}
+
+void PerformAudioPassThruRequest::ProcessAudioPassThruIcon(
+    ApplicationSharedPtr app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  smart_objects::SmartObject& msg_params = (*message_)[strings::msg_params];
+
+  if (msg_params.keyExists(strings::audio_pass_thru_icon)) {
+    smart_objects::SmartObject& icon =
+        msg_params[strings::audio_pass_thru_icon];
+    if (MessageHelper::VerifyImageApplyPath(icon, app, application_manager_) !=
+        mobile_apis::Result::SUCCESS) {
+      LOG4CXX_WARN(
+          logger_,
+          "Invalid audio_pass_thru_icon doesn't exist in the file system");
+    }
+  }
+}
+
+PerformAudioPassThruRequest::AudioPassThruResults
+PerformAudioPassThruRequest::PrepareAudioPassThruResultCodeForResponse(
+    const ResponseInfo& ui_response, const ResponseInfo& tts_response) {
+  mobile_apis::Result::eType result_code = mobile_apis::Result::INVALID_ENUM;
```

Comment:
@dev-gh this method contains both hmi and mobile results. I think it is better to show explicitly hmi and mobile result codes. It looks better.